lib/active_shipping/location.rb

```diff
@@ -42,48 +60,53 @@ def initialize(options = {})
     end
 
     def self.from(object, options = {})
-      return object if object.is_a? ActiveShipping::Location
-      attr_mappings = {
-        :name => [:name],
-        :country => [:country_code, :country],
-        :postal_code => [:postal_code, :zip, :postal],
-        :province => [:province_code, :state_code, :territory_code, :region_code, :province, :state, :territory, :region],
-        :city => [:city, :town],
-        :address1 => [:address1, :address, :street],
-        :address2 => [:address2],
-        :address3 => [:address3],
-        :phone => [:phone, :phone_number],
-        :fax => [:fax, :fax_number],
-        :address_type => [:address_type],
-        :company_name => [:company, :company_name]
-      }
+      return object if object.is_a?(ActiveShipping::Location)
+
       attributes = {}
+
       hash_access = begin
         object[:some_symbol]
         true
       rescue
         false
       end
-      attr_mappings.each do |pair|
-        pair[1].each do |sym|
-          if value = (object[sym] if hash_access) || (object.send(sym) if object.respond_to?(sym) && (!hash_access || !Hash.public_instance_methods.include?(sym.to_s)))
-            attributes[pair[0]] = value
-            break
+
+      ATTRIBUTE_ALIASES.each do |attribute, aliases|
+        aliases.detect do |sym|
```

Comment:
This is a _little_ weird in that `detect` we don't care about the return value, but we do want it to exit early. Test fails if you change it to `each`.